<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Advanced Analyses | Supplementary Materials - OmicsView: omics data analysis through interactive visual analytics</title>
  <meta name="description" content="Chapter 6 Advanced Analyses | Supplementary Materials - OmicsView: omics data analysis through interactive visual analytics" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Advanced Analyses | Supplementary Materials - OmicsView: omics data analysis through interactive visual analytics" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="https://github.com/interactivereport/OmicsView" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Advanced Analyses | Supplementary Materials - OmicsView: omics data analysis through interactive visual analytics" />
  
  
  



<meta name="date" content="2022-02-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="data-upload.html"/>
<link rel="next" href="step-by-step-guide-of-crohns-disease-use-case.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>




<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">OmicsView</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="overview-of-data-in-omicsview.html"><a href="overview-of-data-in-omicsview.html"><i class="fa fa-check"></i><b>1</b> Overview of Data in OmicsView</a>
<ul>
<li class="chapter" data-level="1.1" data-path="overview-of-data-in-omicsview.html"><a href="overview-of-data-in-omicsview.html#projects"><i class="fa fa-check"></i><b>1.1</b> Projects</a></li>
<li class="chapter" data-level="1.2" data-path="overview-of-data-in-omicsview.html"><a href="overview-of-data-in-omicsview.html#samples"><i class="fa fa-check"></i><b>1.2</b> Samples</a></li>
<li class="chapter" data-level="1.3" data-path="overview-of-data-in-omicsview.html"><a href="overview-of-data-in-omicsview.html#comparisons"><i class="fa fa-check"></i><b>1.3</b> Comparisons</a></li>
<li class="chapter" data-level="1.4" data-path="overview-of-data-in-omicsview.html"><a href="overview-of-data-in-omicsview.html#genes"><i class="fa fa-check"></i><b>1.4</b> Genes</a></li>
<li class="chapter" data-level="1.5" data-path="overview-of-data-in-omicsview.html"><a href="overview-of-data-in-omicsview.html#saved-genes-and-comparisons"><i class="fa fa-check"></i><b>1.5</b> Saved Genes and Comparisons</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html"><i class="fa fa-check"></i><b>2</b> Visualize Gene Expression</a>
<ul>
<li class="chapter" data-level="2.1" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#view-gene-expression-from-rna-seq"><i class="fa fa-check"></i><b>2.1</b> View Gene Expression from RNA-Seq</a></li>
<li class="chapter" data-level="2.2" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#change-sample-grouping-in-gene-expression-plot"><i class="fa fa-check"></i><b>2.2</b> Change Sample Grouping in Gene Expression Plot</a></li>
<li class="chapter" data-level="2.3" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#view-gene-expression-from-microarray-data"><i class="fa fa-check"></i><b>2.3</b> View Gene Expression from Microarray Data</a></li>
<li class="chapter" data-level="2.4" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#view-gene-expression-from-multiple-genes"><i class="fa fa-check"></i><b>2.4</b> View Gene Expression from Multiple Genes</a></li>
<li class="chapter" data-level="2.5" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#view-gene-expression-in-heatmap"><i class="fa fa-check"></i><b>2.5</b> View Gene Expression in Heatmap</a></li>
<li class="chapter" data-level="2.6" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#export-genes-and-samples"><i class="fa fa-check"></i><b>2.6</b> Export Genes and Samples</a></li>
<li class="chapter" data-level="2.7" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#similar-comparisons"><i class="fa fa-check"></i><b>2.7</b> Similar comparisons</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#similar-comparisons-go"><i class="fa fa-check"></i><b>2.7.1</b> Similar comparisons (GO)</a></li>
<li class="chapter" data-level="2.7.2" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#similar-comparisons-page"><i class="fa fa-check"></i><b>2.7.2</b> Similar comparisons (PAGE)</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#comparisons-venn-diagrams"><i class="fa fa-check"></i><b>2.8</b> Comparisons Venn Diagrams</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#comparisons-venn-diagram-go"><i class="fa fa-check"></i><b>2.8.1</b> Comparisons Venn Diagram (GO)</a></li>
<li class="chapter" data-level="2.8.2" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#comparisons-venn-diagram-page"><i class="fa fa-check"></i><b>2.8.2</b> Comparisons Venn Diagram (PAGE)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html"><i class="fa fa-check"></i><b>3</b> Visualize Comparison Data</a>
<ul>
<li class="chapter" data-level="3.1" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#dashboard-view-of-comparison"><i class="fa fa-check"></i><b>3.1</b> Dashboard View of Comparison</a></li>
<li class="chapter" data-level="3.2" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#set-dashboard-preference"><i class="fa fa-check"></i><b>3.2</b> Set Dashboard Preference</a></li>
<li class="chapter" data-level="3.3" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#dynamic-filtering-of-comparisons-on-dashboard"><i class="fa fa-check"></i><b>3.3</b> Dynamic Filtering of Comparisons on Dashboard</a></li>
<li class="chapter" data-level="3.4" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#bubble-plot-of-comparisons-associated-with-a-single-gene"><i class="fa fa-check"></i><b>3.4</b> Bubble Plot of Comparisons Associated with a Single Gene</a></li>
<li class="chapter" data-level="3.5" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#data-filter-and-advanced-settings-in-bubble-plot"><i class="fa fa-check"></i><b>3.5</b> Data Filter and Advanced Settings in Bubble Plot</a></li>
<li class="chapter" data-level="3.6" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#bubble-plot-of-sets-of-genes-and-comparisons"><i class="fa fa-check"></i><b>3.6</b> Bubble Plot of Sets of Genes and Comparisons</a></li>
<li class="chapter" data-level="3.7" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#get-significant-genes-from-comparisons"><i class="fa fa-check"></i><b>3.7</b> Get significant genes from comparisons</a></li>
<li class="chapter" data-level="3.8" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#view-significantly-changed-genes-in-bubble-plot"><i class="fa fa-check"></i><b>3.8</b> View Significantly Changed Genes in Bubble Plot</a></li>
<li class="chapter" data-level="3.9" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#volcano-plot-of-a-comparison"><i class="fa fa-check"></i><b>3.9</b> Volcano Plot of a Comparison</a></li>
<li class="chapter" data-level="3.10" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#view-multiple-volcano-plots-together"><i class="fa fa-check"></i><b>3.10</b> View Multiple Volcano Plots Together</a></li>
<li class="chapter" data-level="3.11" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#overlap-comparison-data-to-pathway-graph"><i class="fa fa-check"></i><b>3.11</b> Overlap Comparison Data to Pathway Graph</a></li>
<li class="chapter" data-level="3.12" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#pathway-plot-from-several-comparisons"><i class="fa fa-check"></i><b>3.12</b> Pathway Plot from Several Comparisons</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="functional-enrichment-from-comparisons.html"><a href="functional-enrichment-from-comparisons.html"><i class="fa fa-check"></i><b>4</b> Functional Enrichment from Comparisons</a>
<ul>
<li class="chapter" data-level="4.1" data-path="functional-enrichment-from-comparisons.html"><a href="functional-enrichment-from-comparisons.html#enrichment-from-up-and-down-regulated-genes"><i class="fa fa-check"></i><b>4.1</b> Enrichment from Up and Down Regulated Genes</a></li>
<li class="chapter" data-level="4.2" data-path="functional-enrichment-from-comparisons.html"><a href="functional-enrichment-from-comparisons.html#view-changed-genes-from-a-functional-term-in-volcano-plot"><i class="fa fa-check"></i><b>4.2</b> View Changed Genes from a Functional Term in Volcano Plot</a></li>
<li class="chapter" data-level="4.3" data-path="functional-enrichment-from-comparisons.html"><a href="functional-enrichment-from-comparisons.html#view-enriched-pathways-directly-from-comparison-details"><i class="fa fa-check"></i><b>4.3</b> View Enriched Pathways Directly from Comparison Details</a></li>
<li class="chapter" data-level="4.4" data-path="functional-enrichment-from-comparisons.html"><a href="functional-enrichment-from-comparisons.html#gene-set-enrichment-from-ranked-genes"><i class="fa fa-check"></i><b>4.4</b> Gene Set Enrichment from Ranked Genes</a></li>
<li class="chapter" data-level="4.5" data-path="functional-enrichment-from-comparisons.html"><a href="functional-enrichment-from-comparisons.html#pathway-heatmap-from-comparisons"><i class="fa fa-check"></i><b>4.5</b> Pathway Heatmap From Comparisons</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-upload.html"><a href="data-upload.html"><i class="fa fa-check"></i><b>5</b> Data Upload</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-upload.html"><a href="data-upload.html#data-type"><i class="fa fa-check"></i><b>5.1</b> Data Type</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="data-upload.html"><a href="data-upload.html#supported-data-file-format"><i class="fa fa-check"></i><b>5.1.1</b> Supported Data File Format</a></li>
<li class="chapter" data-level="5.1.2" data-path="data-upload.html"><a href="data-upload.html#expression-data-format"><i class="fa fa-check"></i><b>5.1.2</b> Expression Data Format</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="data-upload.html"><a href="data-upload.html#data-upload-options"><i class="fa fa-check"></i><b>5.2</b> Data Upload options</a></li>
<li class="chapter" data-level="5.3" data-path="data-upload.html"><a href="data-upload.html#source-file-format"><i class="fa fa-check"></i><b>5.3</b> Source File format</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="data-upload.html"><a href="data-upload.html#project-info-file"><i class="fa fa-check"></i><b>5.3.1</b> Project Info file</a></li>
<li class="chapter" data-level="5.3.2" data-path="data-upload.html"><a href="data-upload.html#sample-info-file"><i class="fa fa-check"></i><b>5.3.2</b> Sample Info file</a></li>
<li class="chapter" data-level="5.3.3" data-path="data-upload.html"><a href="data-upload.html#comparison-info-file"><i class="fa fa-check"></i><b>5.3.3</b> Comparison Info file</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="data-upload.html"><a href="data-upload.html#data-formats"><i class="fa fa-check"></i><b>5.4</b> Data Formats</a></li>
<li class="chapter" data-level="5.5" data-path="data-upload.html"><a href="data-upload.html#gene-expression-data"><i class="fa fa-check"></i><b>5.5</b> Gene Expression Data</a></li>
<li class="chapter" data-level="5.6" data-path="data-upload.html"><a href="data-upload.html#comparison-data"><i class="fa fa-check"></i><b>5.6</b> Comparison Data</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="advanced-analyses.html"><a href="advanced-analyses.html"><i class="fa fa-check"></i><b>6</b> Advanced Analyses</a>
<ul>
<li class="chapter" data-level="6.1" data-path="advanced-analyses.html"><a href="advanced-analyses.html#correlation-tools"><i class="fa fa-check"></i><b>6.1</b> Correlation Tools</a></li>
<li class="chapter" data-level="6.2" data-path="advanced-analyses.html"><a href="advanced-analyses.html#pca-analysis"><i class="fa fa-check"></i><b>6.2</b> PCA Analysis</a></li>
<li class="chapter" data-level="6.3" data-path="advanced-analyses.html"><a href="advanced-analyses.html#meta-analysis"><i class="fa fa-check"></i><b>6.3</b> Meta-Analysis</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="advanced-analyses.html"><a href="advanced-analyses.html#meta-analysis-comparisons"><i class="fa fa-check"></i><b>6.3.1</b> Meta-Analysis (Comparisons)</a></li>
<li class="chapter" data-level="6.3.2" data-path="advanced-analyses.html"><a href="advanced-analyses.html#meta-analysis-gene-expression"><i class="fa fa-check"></i><b>6.3.2</b> Meta-Analysis (Gene Expression)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="step-by-step-guide-of-crohns-disease-use-case.html"><a href="step-by-step-guide-of-crohns-disease-use-case.html"><i class="fa fa-check"></i><b>7</b> Step-by-step guide of Crohn’s disease use case</a></li>
<li class="chapter" data-level="8" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>8</b> Methods</a>
<ul>
<li class="chapter" data-level="8.1" data-path="methods.html"><a href="methods.html#data-processing"><i class="fa fa-check"></i><b>8.1</b> Data processing</a></li>
<li class="chapter" data-level="8.2" data-path="methods.html"><a href="methods.html#functional-and-pathway-enrichment"><i class="fa fa-check"></i><b>8.2</b> Functional and Pathway Enrichment</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="methods.html"><a href="methods.html#homer-workflow"><i class="fa fa-check"></i><b>8.2.1</b> HOMER workflow</a></li>
<li class="chapter" data-level="8.2.2" data-path="methods.html"><a href="methods.html#gene-set-enrichment-analysis-gsea-workflow-page"><i class="fa fa-check"></i><b>8.2.2</b> Gene Set Enrichment Analysis (GSEA) workflow (PAGE)</a></li>
<li class="chapter" data-level="8.2.3" data-path="methods.html"><a href="methods.html#meta-analysis-1"><i class="fa fa-check"></i><b>8.2.3</b> Meta-Analysis</a></li>
<li class="chapter" data-level="8.2.4" data-path="methods.html"><a href="methods.html#meta-analysis-statistics"><i class="fa fa-check"></i><b>8.2.4</b> Meta-analysis statistics</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Supplementary Materials - OmicsView: omics data analysis through interactive visual analytics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="advanced-analyses" class="section level1" number="6">
<h1><span class="header-section-number">Chapter 6</span> Advanced Analyses</h1>
<div id="correlation-tools" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Correlation Tools</h2>
<p>Once the user has identified a gene of interest, the user can use correlation tools to find other genes that share similar (or opposite) profiles in terms of gene expression or fold change. First, enter the gene of interest, and samples to be used for correlation. In the example below, we entered OSM gene, and 69 samples involved in Crohn’s disease.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.1.1.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.1.1.jpg" alt="FigureS6.1.1" /></a></p>
<p>Click the plot icon will show scatter plot of the target and the correlated gene.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.1.2.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.1.2.jpg" alt="FigureS6.1.2" /></a></p>
<p>An additional way to plotting the correlation with other genes, is to enable the log2 transformation of the expression.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.1.3.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.1.3.jpg" alt="FigureS6.1.3" /></a></p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.1.4.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.1.4.jpg" alt="FigureS6.1.4" /></a></p>
<p>Now the expression is log transformed.</p>
<p>The examples above are from Correlation Tools using Expression. The interface and usage for Correlation Tools using Comparison is similar.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.1.5.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.1.5.jpg" alt="FigureS6.1.5" /></a></p>
</div>
<div id="pca-analysis" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> PCA Analysis</h2>
<p>Users can select a few samples and use PCA plot to visualize the sample relationships.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.2.1.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.2.1.jpg" alt="FigureS6.2.1" /></a></p>
<p>The system will use FactorMineR package to run PCA analysis and display the results.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.2.2.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.2.2.jpg" alt="FigureS6.2.2" /></a></p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.2.3.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.2.3.jpg" alt="FigureS6.2.3" /></a></p>
<p>The PCA tool can also display results from pre-calculated data (basic PCA, or FactorMineR results).</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.2.4.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.2.4.jpg" alt="FigureS6.2.4" /></a></p>
</div>
<div id="meta-analysis" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> Meta-Analysis</h2>
<p>Meta-Analysis can be used to identify genes that are changed consistently across multiple projects. This can be either implemented by doing Meta-analysis using comparisons or meta-analysis using gene expression.</p>
<div id="meta-analysis-comparisons" class="section level3" number="6.3.1">
<h3><span class="header-section-number">6.3.1</span> Meta-Analysis (Comparisons)</h3>
<p>In the example below, we are looking for the most significant DEGs in Disease vs. Normal comparisons of Crohn’s Disease.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.1.1.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.1.1.jpg" alt="FigureS6.3.1.1" /></a></p>
<p>The system will use the comparison data (logFC, p-value) to compute combined p-value and rank product. It also uses a simple cutoff to get counts of up- and down- regulated genes. This method is fast and can be applied to any comparison data. However, it does not use the individual sample data or considers number of samples in each comparison.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.1.2.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.1.2.jpg" alt="FigureS6.3.1.2" /></a></p>
<p>We can enable filter by percentage up-regulated genes to focus on genes that are consistently up-regulated.</p>
<p>From the output table, the user can select the genes most interesting to them. Some key columns are explained below.</p>
<ul>
<li># of data points: this shows the number of comparisons with valid data for this gene. Although 8 comparisons were entered here, not all genes show up in all experiments. We used minimal 3 data points when we set up display options.</li>
<li>Simple count results. Use the cutoff values we entered (logFC 1, FDR 0.05), the system checks if a gene pass the cutoff for each comparison, and computes percentage up- and down-regulation.</li>
<li>Rank Products results (recommended results for meta-analysis). This is done with the RankProd package. The system rank genes in each comparison using logFC and computes a combined rank and statistical values. The Rank Products columns shows the rank (smaller is more consistent change), log2Fold Change and FDR.</li>
<li>The Combined p-value/FDR (Fisher or Maximum) are computed using the individual p-values with MetaDE package. Note unlike RankProd, these values didn’t consider the direction of the change, so the data need to be interpreted carefully, and we recommend using these p-values together with the simple count values to have a better understanding.</li>
</ul>
<p>The table can be sorted by any column. In the above view, we sorted the data by logFC. The table can also be sorted by # of data points to view the genes with most data. In the example below, we chose 10 up-regulated genes with most data points form the table and used the bubble plot button to create bubble plot. It can be seen that indeed all these genes show consistent up-regulated in the selected comparisons.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.1.3.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.1.3.jpg" alt="FigureS6.3.1.3" /></a></p>
<p>The resulting bubble plot will show all 8 comparisons for each gene.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.1.4.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.1.4.jpg" alt="FigureS6.3.1.4" /></a></p>
<p>We can also create a gene list from the table for up or down gene lists from meta-analysis (using the green button at top of meta-analysis results) and use the list for subsequent study like functional enrichment or generating heatmap. Below is a heatmap using 100 up-regulated genes from meta-analysis, nd its very clear that the diseased samples have an upregulation of these genes.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.1.5.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.1.5.jpg" alt="FigureS6.3.1.5" /></a></p>
</div>
<div id="meta-analysis-gene-expression" class="section level3" number="6.3.2">
<h3><span class="header-section-number">6.3.2</span> Meta-Analysis (Gene Expression)</h3>
<p>In the example below, we use the same comparisons from Crohn’s disease.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.2.1.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.2.1.jpg" alt="FigureS6.3.2.1" /></a></p>
<p>By default, the Rank Product Analysis is checked. Uncheck this if the data is too big for Rank product analysis, or you just need a quick run using limma for each individual comparison.</p>
<p>In the next page, system will display the Case and Control samples for each comparison. At this step, users can edit samples. Users can use this page to remove outliers or add other samples they want to include in the meta-analysis.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.2.2.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.2.2.jpg" alt="FigureS6.3.2.2" /></a></p>
<p>After users review the samples for each comparison, they can go to next step to run the meta-analysis. This step can take a while if the number of samples are large. For the meta-analysis, the system will do the following:</p>
<ol style="list-style-type: decimal">
<li><p>Run limma to get logFC, p-valua and confidence interval for each individual comparison</p></li>
<li><p>Run MetaDE to get combined p-values</p></li>
<li><p>Run Rank Product package for all the expression data to get statistical significance and combined logFC. This step can be very slow is number of samples are large.</p></li>
</ol>
<p>After the analysis is done, the system will show a summary and the result table.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.2.3.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.2.3.jpg" alt="FigureS6.3.2.3" /></a></p>
<p>In the summary, the four comparisons are shown, along with number of genes used for meta-analysis, and number of case and control samples. Because different platforms (especially arrays) have different number of genes present, and some genes may have too low signals to be detected in certain projects, not all genes are present in all the comparisons. In this example, 19,214 genes are present in all comparisons. In some cases, it may be beneficial to remove some comparisons with small number of genes.</p>
<p>In the meta-analysis result table, results from Rank Product are shown (from RankProd package, preferred results to use), followed by effective size method (MetaDE.ES from MetaDE package). The logFC, SE, confidence internal, p-value and FDR for each comparison is shown next. The comparison numbers are the same as those listed in the summary table above. The gene list can be sorted by RankProd (most significant changes) or by logFC_RP (up or down-regulated).</p>
<p>From this analysis, we can create Forest plot for each gene by click the icon near the gene symbol.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.2.4.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.2.4.jpg" alt="FigureS6.3.2.4" /></a></p>
<p>In the Forest plot, results from individual comparison and combined data are shown. The confidence internal for individual comparison is based on limma output. The confidence internal for combined results is based on p-value from RankProd (or ES method when RankProd is not run).</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.2.5.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS6.3.2.5.jpg" alt="FigureS6.3.2.5" /></a></p>
<p>The plot can be saved as SVG or PDF. The summary table is shown below the plot.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="data-upload.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="step-by-step-guide-of-crohns-disease-use-case.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/Tutorial_6.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
},
"pandoc_args": ["--lua-filter=scholarly-metadata.lua", "--lua-filter=author-info-blocks.lua"]
});
});
</script>

</body>

</html>
