<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Visualize Gene Expression | Supplementary Materials - OmicsView: omics data analysis through interactive visual analytics</title>
  <meta name="description" content="Chapter 2 Visualize Gene Expression | Supplementary Materials - OmicsView: omics data analysis through interactive visual analytics" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Visualize Gene Expression | Supplementary Materials - OmicsView: omics data analysis through interactive visual analytics" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="https://github.com/interactivereport/OmicsView" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Visualize Gene Expression | Supplementary Materials - OmicsView: omics data analysis through interactive visual analytics" />
  
  
  



<meta name="date" content="2022-02-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="overview-of-data-in-omicsview.html"/>
<link rel="next" href="visualize-comparison-data.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>




<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">OmicsView</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="chapter" data-level="1" data-path="overview-of-data-in-omicsview.html"><a href="overview-of-data-in-omicsview.html"><i class="fa fa-check"></i><b>1</b> Overview of Data in OmicsView</a>
<ul>
<li class="chapter" data-level="1.1" data-path="overview-of-data-in-omicsview.html"><a href="overview-of-data-in-omicsview.html#projects"><i class="fa fa-check"></i><b>1.1</b> Projects</a></li>
<li class="chapter" data-level="1.2" data-path="overview-of-data-in-omicsview.html"><a href="overview-of-data-in-omicsview.html#samples"><i class="fa fa-check"></i><b>1.2</b> Samples</a></li>
<li class="chapter" data-level="1.3" data-path="overview-of-data-in-omicsview.html"><a href="overview-of-data-in-omicsview.html#comparisons"><i class="fa fa-check"></i><b>1.3</b> Comparisons</a></li>
<li class="chapter" data-level="1.4" data-path="overview-of-data-in-omicsview.html"><a href="overview-of-data-in-omicsview.html#genes"><i class="fa fa-check"></i><b>1.4</b> Genes</a></li>
<li class="chapter" data-level="1.5" data-path="overview-of-data-in-omicsview.html"><a href="overview-of-data-in-omicsview.html#saved-genes-and-comparisons"><i class="fa fa-check"></i><b>1.5</b> Saved Genes and Comparisons</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html"><i class="fa fa-check"></i><b>2</b> Visualize Gene Expression</a>
<ul>
<li class="chapter" data-level="2.1" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#view-gene-expression-from-rna-seq"><i class="fa fa-check"></i><b>2.1</b> View Gene Expression from RNA-Seq</a></li>
<li class="chapter" data-level="2.2" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#change-sample-grouping-in-gene-expression-plot"><i class="fa fa-check"></i><b>2.2</b> Change Sample Grouping in Gene Expression Plot</a></li>
<li class="chapter" data-level="2.3" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#view-gene-expression-from-microarray-data"><i class="fa fa-check"></i><b>2.3</b> View Gene Expression from Microarray Data</a></li>
<li class="chapter" data-level="2.4" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#view-gene-expression-from-multiple-genes"><i class="fa fa-check"></i><b>2.4</b> View Gene Expression from Multiple Genes</a></li>
<li class="chapter" data-level="2.5" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#view-gene-expression-in-heatmap"><i class="fa fa-check"></i><b>2.5</b> View Gene Expression in Heatmap</a></li>
<li class="chapter" data-level="2.6" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#export-genes-and-samples"><i class="fa fa-check"></i><b>2.6</b> Export Genes and Samples</a></li>
<li class="chapter" data-level="2.7" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#similar-comparisons"><i class="fa fa-check"></i><b>2.7</b> Similar comparisons</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#similar-comparisons-go"><i class="fa fa-check"></i><b>2.7.1</b> Similar comparisons (GO)</a></li>
<li class="chapter" data-level="2.7.2" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#similar-comparisons-page"><i class="fa fa-check"></i><b>2.7.2</b> Similar comparisons (PAGE)</a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#comparisons-venn-diagrams"><i class="fa fa-check"></i><b>2.8</b> Comparisons Venn Diagrams</a>
<ul>
<li class="chapter" data-level="2.8.1" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#comparisons-venn-diagram-go"><i class="fa fa-check"></i><b>2.8.1</b> Comparisons Venn Diagram (GO)</a></li>
<li class="chapter" data-level="2.8.2" data-path="visualize-gene-expression.html"><a href="visualize-gene-expression.html#comparisons-venn-diagram-page"><i class="fa fa-check"></i><b>2.8.2</b> Comparisons Venn Diagram (PAGE)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html"><i class="fa fa-check"></i><b>3</b> Visualize Comparison Data</a>
<ul>
<li class="chapter" data-level="3.1" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#dashboard-view-of-comparison"><i class="fa fa-check"></i><b>3.1</b> Dashboard View of Comparison</a></li>
<li class="chapter" data-level="3.2" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#set-dashboard-preference"><i class="fa fa-check"></i><b>3.2</b> Set Dashboard Preference</a></li>
<li class="chapter" data-level="3.3" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#dynamic-filtering-of-comparisons-on-dashboard"><i class="fa fa-check"></i><b>3.3</b> Dynamic Filtering of Comparisons on Dashboard</a></li>
<li class="chapter" data-level="3.4" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#bubble-plot-of-comparisons-associated-with-a-single-gene"><i class="fa fa-check"></i><b>3.4</b> Bubble Plot of Comparisons Associated with a Single Gene</a></li>
<li class="chapter" data-level="3.5" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#data-filter-and-advanced-settings-in-bubble-plot"><i class="fa fa-check"></i><b>3.5</b> Data Filter and Advanced Settings in Bubble Plot</a></li>
<li class="chapter" data-level="3.6" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#bubble-plot-of-sets-of-genes-and-comparisons"><i class="fa fa-check"></i><b>3.6</b> Bubble Plot of Sets of Genes and Comparisons</a></li>
<li class="chapter" data-level="3.7" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#get-significant-genes-from-comparisons"><i class="fa fa-check"></i><b>3.7</b> Get significant genes from comparisons</a></li>
<li class="chapter" data-level="3.8" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#view-significantly-changed-genes-in-bubble-plot"><i class="fa fa-check"></i><b>3.8</b> View Significantly Changed Genes in Bubble Plot</a></li>
<li class="chapter" data-level="3.9" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#volcano-plot-of-a-comparison"><i class="fa fa-check"></i><b>3.9</b> Volcano Plot of a Comparison</a></li>
<li class="chapter" data-level="3.10" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#view-multiple-volcano-plots-together"><i class="fa fa-check"></i><b>3.10</b> View Multiple Volcano Plots Together</a></li>
<li class="chapter" data-level="3.11" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#overlap-comparison-data-to-pathway-graph"><i class="fa fa-check"></i><b>3.11</b> Overlap Comparison Data to Pathway Graph</a></li>
<li class="chapter" data-level="3.12" data-path="visualize-comparison-data.html"><a href="visualize-comparison-data.html#pathway-plot-from-several-comparisons"><i class="fa fa-check"></i><b>3.12</b> Pathway Plot from Several Comparisons</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="functional-enrichment-from-comparisons.html"><a href="functional-enrichment-from-comparisons.html"><i class="fa fa-check"></i><b>4</b> Functional Enrichment from Comparisons</a>
<ul>
<li class="chapter" data-level="4.1" data-path="functional-enrichment-from-comparisons.html"><a href="functional-enrichment-from-comparisons.html#enrichment-from-up-and-down-regulated-genes"><i class="fa fa-check"></i><b>4.1</b> Enrichment from Up and Down Regulated Genes</a></li>
<li class="chapter" data-level="4.2" data-path="functional-enrichment-from-comparisons.html"><a href="functional-enrichment-from-comparisons.html#view-changed-genes-from-a-functional-term-in-volcano-plot"><i class="fa fa-check"></i><b>4.2</b> View Changed Genes from a Functional Term in Volcano Plot</a></li>
<li class="chapter" data-level="4.3" data-path="functional-enrichment-from-comparisons.html"><a href="functional-enrichment-from-comparisons.html#view-enriched-pathways-directly-from-comparison-details"><i class="fa fa-check"></i><b>4.3</b> View Enriched Pathways Directly from Comparison Details</a></li>
<li class="chapter" data-level="4.4" data-path="functional-enrichment-from-comparisons.html"><a href="functional-enrichment-from-comparisons.html#gene-set-enrichment-from-ranked-genes"><i class="fa fa-check"></i><b>4.4</b> Gene Set Enrichment from Ranked Genes</a></li>
<li class="chapter" data-level="4.5" data-path="functional-enrichment-from-comparisons.html"><a href="functional-enrichment-from-comparisons.html#pathway-heatmap-from-comparisons"><i class="fa fa-check"></i><b>4.5</b> Pathway Heatmap From Comparisons</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-upload.html"><a href="data-upload.html"><i class="fa fa-check"></i><b>5</b> Data Upload</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-upload.html"><a href="data-upload.html#data-type"><i class="fa fa-check"></i><b>5.1</b> Data Type</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="data-upload.html"><a href="data-upload.html#supported-data-file-format"><i class="fa fa-check"></i><b>5.1.1</b> Supported Data File Format</a></li>
<li class="chapter" data-level="5.1.2" data-path="data-upload.html"><a href="data-upload.html#expression-data-format"><i class="fa fa-check"></i><b>5.1.2</b> Expression Data Format</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="data-upload.html"><a href="data-upload.html#data-upload-options"><i class="fa fa-check"></i><b>5.2</b> Data Upload options</a></li>
<li class="chapter" data-level="5.3" data-path="data-upload.html"><a href="data-upload.html#source-file-format"><i class="fa fa-check"></i><b>5.3</b> Source File format</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="data-upload.html"><a href="data-upload.html#project-info-file"><i class="fa fa-check"></i><b>5.3.1</b> Project Info file</a></li>
<li class="chapter" data-level="5.3.2" data-path="data-upload.html"><a href="data-upload.html#sample-info-file"><i class="fa fa-check"></i><b>5.3.2</b> Sample Info file</a></li>
<li class="chapter" data-level="5.3.3" data-path="data-upload.html"><a href="data-upload.html#comparison-info-file"><i class="fa fa-check"></i><b>5.3.3</b> Comparison Info file</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="data-upload.html"><a href="data-upload.html#data-formats"><i class="fa fa-check"></i><b>5.4</b> Data Formats</a></li>
<li class="chapter" data-level="5.5" data-path="data-upload.html"><a href="data-upload.html#gene-expression-data"><i class="fa fa-check"></i><b>5.5</b> Gene Expression Data</a></li>
<li class="chapter" data-level="5.6" data-path="data-upload.html"><a href="data-upload.html#comparison-data"><i class="fa fa-check"></i><b>5.6</b> Comparison Data</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="advanced-analyses.html"><a href="advanced-analyses.html"><i class="fa fa-check"></i><b>6</b> Advanced Analyses</a>
<ul>
<li class="chapter" data-level="6.1" data-path="advanced-analyses.html"><a href="advanced-analyses.html#correlation-tools"><i class="fa fa-check"></i><b>6.1</b> Correlation Tools</a></li>
<li class="chapter" data-level="6.2" data-path="advanced-analyses.html"><a href="advanced-analyses.html#pca-analysis"><i class="fa fa-check"></i><b>6.2</b> PCA Analysis</a></li>
<li class="chapter" data-level="6.3" data-path="advanced-analyses.html"><a href="advanced-analyses.html#meta-analysis"><i class="fa fa-check"></i><b>6.3</b> Meta-Analysis</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="advanced-analyses.html"><a href="advanced-analyses.html#meta-analysis-comparisons"><i class="fa fa-check"></i><b>6.3.1</b> Meta-Analysis (Comparisons)</a></li>
<li class="chapter" data-level="6.3.2" data-path="advanced-analyses.html"><a href="advanced-analyses.html#meta-analysis-gene-expression"><i class="fa fa-check"></i><b>6.3.2</b> Meta-Analysis (Gene Expression)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="step-by-step-guide-of-crohns-disease-use-case.html"><a href="step-by-step-guide-of-crohns-disease-use-case.html"><i class="fa fa-check"></i><b>7</b> Step-by-step guide of Crohn’s disease use case</a></li>
<li class="chapter" data-level="8" data-path="methods.html"><a href="methods.html"><i class="fa fa-check"></i><b>8</b> Methods</a>
<ul>
<li class="chapter" data-level="8.1" data-path="methods.html"><a href="methods.html#data-processing"><i class="fa fa-check"></i><b>8.1</b> Data processing</a></li>
<li class="chapter" data-level="8.2" data-path="methods.html"><a href="methods.html#functional-and-pathway-enrichment"><i class="fa fa-check"></i><b>8.2</b> Functional and Pathway Enrichment</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="methods.html"><a href="methods.html#homer-workflow"><i class="fa fa-check"></i><b>8.2.1</b> HOMER workflow</a></li>
<li class="chapter" data-level="8.2.2" data-path="methods.html"><a href="methods.html#gene-set-enrichment-analysis-gsea-workflow-page"><i class="fa fa-check"></i><b>8.2.2</b> Gene Set Enrichment Analysis (GSEA) workflow (PAGE)</a></li>
<li class="chapter" data-level="8.2.3" data-path="methods.html"><a href="methods.html#meta-analysis-1"><i class="fa fa-check"></i><b>8.2.3</b> Meta-Analysis</a></li>
<li class="chapter" data-level="8.2.4" data-path="methods.html"><a href="methods.html#meta-analysis-statistics"><i class="fa fa-check"></i><b>8.2.4</b> Meta-analysis statistics</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Supplementary Materials - OmicsView: omics data analysis through interactive visual analytics</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="visualize-gene-expression" class="section level1" number="2">
<h1><span class="header-section-number">Chapter 2</span> Visualize Gene Expression</h1>
<p>For each gene, Users can view its expression levels across multiple samples. Most data in OmicsView are from microarray data, consisting of more than 50K samples, and nearly 3000 samples are from RNA-Seq.</p>
<div id="view-gene-expression-from-rna-seq" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> View Gene Expression from RNA-Seq</h2>
<p>Choose the Gene Expression from RNA-Seq -&gt; Single Gene from left menu, and enter the official symbol of gene. Alternatively, in the gene details page, click View Gene Expression link.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.1.1.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.1.1.jpg" alt="FigureS2.1.1" /></a></p>
<p>As an optional step, Users can choose what sample attributes to pass to the plot and use data filter to choose only a subset of data points.
The Data Filter can be very useful if there are too many data points and you can focus on a few diseases or tissue types.
The screenshots below show default boxplot showing all diseases, and by after filtering for few diseases.
 
Plot showing all disease states (scroll to view other diseases). 5000 randomly selected data points out of 9538 are shown.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.1.2.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.1.2.jpg" alt="FigureS2.1.2" /></a></p>
<p>Plot after data filtering was applied to select a few diseases. Now 410 out of 9538 data points are shown. The data filter pop-up window was shown to the left of the box plot in the screen shot below.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.1.3.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.1.3.jpg" alt="FigureS2.1.3" /></a></p>
</div>
<div id="change-sample-grouping-in-gene-expression-plot" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> Change Sample Grouping in Gene Expression Plot</h2>
<p>The boxplot is created using CanvasXpress ( <a href="https://canvasxpress.org" class="uri">https://canvasxpress.org</a> )plug-in, and sample grouping and coloring can be customized by the user. In the example below, we show how sample grouping can be changed.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.2.1.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.2.1.jpg" alt="FigureS2.2.1" /></a></p>
<p>Once the sample grouping was changed to tissue, the box plot shows that insulin is only expressed in pancreatic islets. Users can also change how the data points are colored (default is by disease).</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.2.2.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.2.2.jpg" alt="FigureS2.2.2" /></a></p>
</div>
<div id="view-gene-expression-from-microarray-data" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> View Gene Expression from Microarray Data</h2>
<p>The way to view microarray data is very similar to RNA-Seq data. However, since the expression values from different array platforms are typically not directly comparable, the system by default will choose the platform with the largest number of data points. The user can override this filter if needed.
In addition, we recommend the user to add data filter because typically there are too many data points from microarray data. Data filters will help the user to focus on the most important tissues or diseases. If there are still more than 5000 data points after data filer, only the first 5000 are shown in the boxplot.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.3.1.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.3.1.jpg" alt="FigureS2.3.1" /></a></p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.3.2.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.3.2.jpg" alt="FigureS2.3.2" /></a></p>
<p>The sample grouping and coloring can be changed by the user. See the section 2.2 for details.</p>
</div>
<div id="view-gene-expression-from-multiple-genes" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> View Gene Expression from Multiple Genes</h2>
<p>Users can view multiple genes on the same boxplot. The interface is similar to single gene, here you need to enter a list of gene symbols. Users can also load the saved gene lists from your collection. In multiple gene plot, you want to filter the data to come up with a reasonable number of data points.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.4.1.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.4.1.jpg" alt="FigureS2.4.1" /></a></p>
<p>From multiple gene plot, you can use the built-in function of CanvassXpress to change grouping and coloring.</p>
<p>In the example plot above, we can change the setting to use group by to distinguish Crohn’s disease samples and control samples.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.4.2.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.4.2.jpg" alt="FigureS2.4.2" /></a></p>
<p>In the updated plot above, it’s easy to see that many of these genes are expressed slightly higher in disease (red) vs. control (blue).</p>
</div>
<div id="view-gene-expression-in-heatmap" class="section level2" number="2.5">
<h2><span class="header-section-number">2.5</span> View Gene Expression in Heatmap</h2>
<p>Heatmap can be useful to visualize gene profiles from multiple samples. It can also provide information about how genes and samples cluster. This data is from PMID: 25003194</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.5.1.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.5.1.jpg" alt="FigureS2.5.1" /></a></p>
<p>Users can enter genes and samples in the box, or load pre-saved genes and samples quickly from your collection. Be default, we will log2 transform the gene expression data, perform scaling of the data across samples for each gene, and limit the scaled value to -3 to 3 before displaying the data in heatmap. This works well in most situations. However, advanced users can change the options. For example, if you want to keep the order of samples as you entered, just uncheck “Enable Clustering Samples.” 
The heatmap is rendered by CanvasXpress. Users can change the plot size if needed.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.5.2.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.5.2.jpg" alt="FigureS2.5.2" /></a></p>
<p>In the example heatmap, we entered a few significantly changed genes in Crohn’s disease, and choose a few disease and control samples. As shown in the heatmap, most of these genes show distinct patterns between disease and control.
One of the samples GSM1598422 is labeled as disease, but its gene expression signature match those with normal control very well. Therefore from heatmap clustering, we can decide that this sample is likely an outlier, the sample may be mislabeled, or this patient had very different gene expression patterns form all other patients.</p>
</div>
<div id="export-genes-and-samples" class="section level2" number="2.6">
<h2><span class="header-section-number">2.6</span> Export Genes and Samples</h2>
<p>This is a useful feature to download raw (counts) and processed (TPM values) expression matrices from the datasets. This gives the user the opportunity to do any downstream analysis on the raw reads independently.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.6.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.6.jpg" alt="FigureS2.6" /></a></p>
<p>For this functionality, the samples need to belong to the same platform type and the requested limit is to 100 genes and 20 samples. The files are exported as csv and can be loaded and used by the user to perform their own downstream analysis.</p>
</div>
<div id="similar-comparisons" class="section level2" number="2.7">
<h2><span class="header-section-number">2.7</span> Similar comparisons</h2>
<p>For a given comparison, this functionality helps identify other similar comparisons.</p>
<div id="similar-comparisons-go" class="section level3" number="2.7.1">
<h3><span class="header-section-number">2.7.1</span> Similar comparisons (GO)</h3>
<p>To identify similar comparisons based on GO terms in up or downregulated genes.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.7.1.1.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.7.1.1.jpg" alt="FigureS2.7.1.1" /></a></p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.7.1.2.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.7.1.2.jpg" alt="FigureS2.7.1.2" /></a></p>
</div>
<div id="similar-comparisons-page" class="section level3" number="2.7.2">
<h3><span class="header-section-number">2.7.2</span> Similar comparisons (PAGE)</h3>
<p>Similarly, the similar comparisons can be identified using PAGE results</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.7.2.1.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.7.2.1.jpg" alt="FigureS2.7.2.1" /></a></p>
<p>A list of comparisons that were found similar to the target comparison are displayed. They can be sorted on different attributed like # Terms in Comparison, # Overlaps and. Enrichment score</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.7.2.2.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.7.2.2.jpg" alt="FigureS2.7.2.2" /></a></p>
</div>
</div>
<div id="comparisons-venn-diagrams" class="section level2" number="2.8">
<h2><span class="header-section-number">2.8</span> Comparisons Venn Diagrams</h2>
<p>A good way to identify similarities and difference between different comparisons is by using the Comparisons Venn Diagram functionality. There are two ways to use this tool: By GO terms and PAGE results.</p>
<div id="comparisons-venn-diagram-go" class="section level3" number="2.8.1">
<h3><span class="header-section-number">2.8.1</span> Comparisons Venn Diagram (GO)</h3>
<p>For identifying similar GO terms between comparisons either in the Up and Down regulated categories, this functionality can be used. In this example, 3 comparisons from Psoriasis are probed, the user also has the option of probing 2 comparisons by leaving the third field empty.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.8.1.1.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.8.1.1.jpg" alt="FigureS2.8.1.1" /></a></p>
<p>Upon finishing, an overlap summary is generated. This contains a Venn Diagram of the intersection of all the GO terms between the 3 comparisons. The count number can be clicked, as shown below, to view which terms are included.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.8.1.2.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.8.1.2.jpg" alt="FigureS2.8.1.2" /></a></p>
<p>Furthermore, there is another section below this where pairs of the comparisons can be probed. Here, just A &amp; B are displayed.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.8.1.3.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.8.1.3.jpg" alt="FigureS2.8.1.3" /></a></p>
</div>
<div id="comparisons-venn-diagram-page" class="section level3" number="2.8.2">
<h3><span class="header-section-number">2.8.2</span> Comparisons Venn Diagram (PAGE)</h3>
<p>Similarly, the user can probe the similarities between PAGE results or 2 or 3 comparisons. In this example we probe comparisons from Psoriasis again.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.8.2.1.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.8.2.1.jpg" alt="FigureS2.8.2.1" /></a></p>
<p>Upon finishing an overlap summary is generated which shows the intersection of the PAGE results between the comparisons.</p>
<p><a href="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.8.2.2.jpg"><img src="https://interactivereport.github.io/OmicsView/tutorial/figures/FigureS2.8.2.2.jpg" alt="FigureS2.8.2.2" /></a></p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="overview-of-data-in-omicsview.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visualize-comparison-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/Tutorial_2.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
},
"pandoc_args": ["--lua-filter=scholarly-metadata.lua", "--lua-filter=author-info-blocks.lua"]
});
});
</script>

</body>

</html>
